internal/credential/vault/service_list.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/credential"
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// List lists both generic and SSH credential libraries.
+// Supports the following options:
+//   - credential.WithLimit
+//   - credential.WithStartPageAfterItem
+func (s *LibraryService) List(ctx context.Context, credentialStoreId string, opts ...credential.Option) ([]credential.Library, error) {
+	const op = "vault.(*LibraryService).List"
+	opt, err := credential.GetOpts(opts...)
```

Comment:
I'd prefer to see the parameters validated vs relying on other funcs to validate
them. (this preference applies to all the receiver funcs in this file I believe)

Reply:
Good shout, I'll add validation.